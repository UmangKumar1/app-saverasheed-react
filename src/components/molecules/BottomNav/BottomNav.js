import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled, { css, cx } from 'react-emotion'
import { map, startsWith } from 'ramda'

import theme from '../../../theme'
import { noop } from '../../../util/ramda-extra'
import ButtonLink from '../../atoms/ButtonLink'
import BackToMap from './../../icons/ButtonBackToMap'
import CenterMap from './../../icons/ButtonCenterMap'
import CurrentQuest from './../../icons/ButtonCurrentQuest'
import NextObjective from './../../icons/ButtonNextObjective'
import QuestLog from './../../icons/ButtonQuestLog'
import Select from './../../icons/ButtonSelect'
import Profile from './../../icons/ButtonProfile'

const { object, string } = PropTypes

noop(BackToMap, CurrentQuest, NextObjective, Select, CenterMap)

const {
  palette: {
    tertiary: { main, contrastText },
    secondary: { main: secondary },
  },
} = theme

const BottomNavRoot = styled('div')({
  width: '100%',
  position: 'fixed',
  bottom: 0,
  backgroundColor: main,
  color: contrastText,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-around',
  height: 75,
  zIndex: 2,
})

const getNavItems = ({ activeQuestSlug }) => {
  const left = {
    label: 'Profile',
    Icon: Profile,
    key: 'profile',
    to: '/profile',
  }
  const middle = {
    label: 'Map',
    Icon: BackToMap,
    key: '',
    to: '/',
  }
  const right = {
    label: 'Quest Log',
    Icon: QuestLog,
    key: 'quest-log',
    to: activeQuestSlug ? `/quest-log/${activeQuestSlug}` : '/quest-log',
  }
  return [left, middle, right]
}

const btnWrapperCx = css({ backgroundColor: 'inherit' })
const selectedBtnWrapperCx = css({ color: secondary })
const btnCx = css({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  flex: 1,
  backgroundColor: 'inherit',
})

class BottomNav extends Component {
  static propTypes = {
    location: object.isRequired,
    activeQuestSlug: string,
  }

  state = { selected: null }

  handleChange = selected => this.setState({ selected })

  render() {
    const {
      location: { pathname },
      activeQuestSlug,
    } = this.props

    const navItems = getNavItems({ activeQuestSlug })

    return (
      <BottomNavRoot>
        {map(
          ({ label, Icon, key, to, onClick }) => (
            <ButtonLink
              id={`bottom-nav-${key}`}
              to={to}
              key={key}
              onClick={onClick}
              btnWrapperClassName={cx(btnWrapperCx, {
                [selectedBtnWrapperCx]:
                  key === ''
                    ? pathname === '/'
                    : startsWith(`/${key}`, pathname),
              })}
              btnClassName={btnCx}
            >
              <Icon size={48} />
              {label}
            </ButtonLink>
          ),
          navItems,
        )}
      </BottomNavRoot>
    )
  }
}

export default BottomNav
