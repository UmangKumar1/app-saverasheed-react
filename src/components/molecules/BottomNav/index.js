export { default, default as BottomNav } from './BottomNav'
