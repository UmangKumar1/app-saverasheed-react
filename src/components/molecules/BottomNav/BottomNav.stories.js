import React from 'react'
import { storiesOf } from '@storybook/react'
import StoryRouter from 'storybook-react-router'

import BottomNav from '.'
import { sampleLocation } from '../../../fixtures/router'

storiesOf('BottomNav', module)
  .addDecorator(StoryRouter())
  .add('default', () => (
    <BottomNav location={sampleLocation} activeQuestSlug="foo-bar">
      BottomNav
    </BottomNav>
  ))
