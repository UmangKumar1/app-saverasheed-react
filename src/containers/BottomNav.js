import { connect } from 'react-redux'
import { getLocation } from '../modules/common/selectors'
import { getActiveQuestSlug } from '../modules/quest/selectors'

import { subscribe } from '../util/reselect'
import BottomNav from '../components/molecules/BottomNav'

const mapStateToProps = subscribe({
  getLocation,
  getActiveQuestSlug,
})

const actions = {}

export default connect(
  mapStateToProps,
  actions,
)(BottomNav)
