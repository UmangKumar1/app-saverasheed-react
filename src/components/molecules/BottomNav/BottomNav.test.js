import React from 'react'
import { shallow } from 'enzyme'

import BottomNav from '.'
import { sampleLocation } from '../../../fixtures/router'

const wrap = (props = {}) => shallow(<BottomNav {...props} />)

describe('BottomNav', () => {
  it('renders without crashing', () => {
    const wrapper = wrap({
      location: sampleLocation,
      activeQuestSlug: 'foo-bar',
    })
    expect(wrapper).toMatchSnapshot()
  })
})
