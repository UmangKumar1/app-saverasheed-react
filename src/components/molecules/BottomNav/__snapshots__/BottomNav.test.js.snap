// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNav renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BottomNav
    activeQuestSlug="foo-bar"
    location={
      Object {
        "pathname": "/",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ButtonLink
          btnClassName="css-rkff0y"
          btnWrapperClassName="css-1dww627"
          id="bottom-nav-profile"
          to="/profile"
        >
          <ButtonProfile
            size={48}
          />
          Profile
        </ButtonLink>,
        <ButtonLink
          btnClassName="css-rkff0y"
          btnWrapperClassName="css-1b3b6dh"
          id="bottom-nav-"
          to="/"
        >
          <ButtonBackToMap
            size={48}
          />
          Map
        </ButtonLink>,
        <ButtonLink
          btnClassName="css-rkff0y"
          btnWrapperClassName="css-1dww627"
          id="bottom-nav-quest-log"
          to="/quest-log/foo-bar"
        >
          <ButtonQuestLog
            size={48}
          />
          Quest Log
        </ButtonLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "profile",
        "nodeType": "function",
        "props": Object {
          "btnClassName": "css-rkff0y",
          "btnWrapperClassName": "css-1dww627",
          "children": Array [
            <ButtonProfile
              size={48}
            />,
            "Profile",
          ],
          "id": "bottom-nav-profile",
          "onClick": undefined,
          "to": "/profile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 48,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Profile",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "",
        "nodeType": "function",
        "props": Object {
          "btnClassName": "css-rkff0y",
          "btnWrapperClassName": "css-1b3b6dh",
          "children": Array [
            <ButtonBackToMap
              size={48}
            />,
            "Map",
          ],
          "id": "bottom-nav-",
          "onClick": undefined,
          "to": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 48,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Map",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "quest-log",
        "nodeType": "function",
        "props": Object {
          "btnClassName": "css-rkff0y",
          "btnWrapperClassName": "css-1dww627",
          "children": Array [
            <ButtonQuestLog
              size={48}
            />,
            "Quest Log",
          ],
          "id": "bottom-nav-quest-log",
          "onClick": undefined,
          "to": "/quest-log/foo-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 48,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Quest Log",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ButtonLink
            btnClassName="css-rkff0y"
            btnWrapperClassName="css-1dww627"
            id="bottom-nav-profile"
            to="/profile"
          >
            <ButtonProfile
              size={48}
            />
            Profile
          </ButtonLink>,
          <ButtonLink
            btnClassName="css-rkff0y"
            btnWrapperClassName="css-1b3b6dh"
            id="bottom-nav-"
            to="/"
          >
            <ButtonBackToMap
              size={48}
            />
            Map
          </ButtonLink>,
          <ButtonLink
            btnClassName="css-rkff0y"
            btnWrapperClassName="css-1dww627"
            id="bottom-nav-quest-log"
            to="/quest-log/foo-bar"
          >
            <ButtonQuestLog
              size={48}
            />
            Quest Log
          </ButtonLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "profile",
          "nodeType": "function",
          "props": Object {
            "btnClassName": "css-rkff0y",
            "btnWrapperClassName": "css-1dww627",
            "children": Array [
              <ButtonProfile
                size={48}
              />,
              "Profile",
            ],
            "id": "bottom-nav-profile",
            "onClick": undefined,
            "to": "/profile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "size": 48,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Profile",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "",
          "nodeType": "function",
          "props": Object {
            "btnClassName": "css-rkff0y",
            "btnWrapperClassName": "css-1b3b6dh",
            "children": Array [
              <ButtonBackToMap
                size={48}
              />,
              "Map",
            ],
            "id": "bottom-nav-",
            "onClick": undefined,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "size": 48,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Map",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "quest-log",
          "nodeType": "function",
          "props": Object {
            "btnClassName": "css-rkff0y",
            "btnWrapperClassName": "css-1dww627",
            "children": Array [
              <ButtonQuestLog
                size={48}
              />,
              "Quest Log",
            ],
            "id": "bottom-nav-quest-log",
            "onClick": undefined,
            "to": "/quest-log/foo-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "size": 48,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Quest Log",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
